lib/policies/load-balancing.js

```diff
@@ -319,39 +336,32 @@ TokenAwareIterator.prototype.iterate = function (callback) {
 };
 
 TokenAwareIterator.prototype.computeNext = function () {
-  //return local replicas
-  var host = this.nextLocalReplica();
-  if (host) {
-    return {value: host, done: !host};
+  var host;
+  if (this.localReplicaIndex < this.localReplicas.length) {
+    host = this.localReplicas[(this.startIndex + (this.localReplicaIndex++)) % this.localReplicas.length];
+    return { value: host, done: false };
   }
-  //return child hosts
-  var item = this.childIterator.next();
-  if (!item.done) {
+  this.startIndex = -1;
+  // Return hosts from child policy
+  var item;
+  while ((item = this.childIterator.next()) && !item.done) {
     if (this.replicaMap[item.value.address]) {
-      return this.computeNext();
+      // Avoid yielding replicas from the child load balancing policy query plan
```

Comment:
I was thinking that the child policy can still control if the driver should create connections to it by setting the distance, checking the host datacenter. In that way, the child policy implementer doesn't have to include all the nodes... but in the case that the child policy doesn't correctly handles distance for those replicas, it could lead to unexpected bebaviour (unwanted inter-dc traffic).
I will remove that logic.